pytorch_lightning/loops/base.py

```diff
@@ -49,6 +49,7 @@ class Loop(ABC, Generic[T]):
 
     def __init__(self) -> None:
         self.restarting = False
+        self.is_fresh_start_epoch: bool = True
```

Comment:
i do not think we should be introducing this attribute to the base loop interface. not all loops implement epoch-based semantics.

Reply:
Agree, this look like a hack and it won't generalize to all loops.